src/app/modules/main/chat_section/chat_content/messages/module.nim

```diff
@@ -661,13 +662,21 @@ proc switchToMessage*(self: Module, messageId: string) =
     self.controller.setSearchedMessageId(messageId)
 
 method scrollToMessage*(self: Module, messageId: string) =
-  if(messageId == ""):
+  if messageId == "":
     return
 
-  if(self.view.getMessageSearchOngoing()):
+  if self.view.getMessageSearchOngoing():
     return
 
+  self.controller.asyncGetMessageById(messageId, GetMessageByIdCallReason.ScrollTomessage)
```

Comment:
I believe more practical way of doing this would be to use a `requestID`, rather than `callReason`. There are many message modules which may trigger `asyncGetMessageById` with `ScrollTomessage` reason at the same time. If that's the case, then it would break the loading animation in some modules.

```suggestion
  self.getMessageRequestID = self.controller.asyncGetMessageById(messageId)
```

This ID could be a UUID generated by service at the time of the call, and then propagated with the response.

Reply:
hmm, good idea!